### PR TITLE
Add versions endpoint for about dialog in fronts

### PR DESCRIPTION
**Pull request by Tristan-WorkGH**

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/RemoteServicesProperties.java [hunk 1]
```diff
@@ -4,33 +4,50 @@
...
 /**
  * @author David Braquart <david.braquart at rte-france.com>
  */
+@Validated
 @Component
 @ConfigurationProperties(prefix = "gridsuite")
+@AllArgsConstructor
+@NoArgsConstructor
 @Data
 public class RemoteServicesProperties {
```
Comment (flomillot) [inline src/main/java/org/gridsuite/study/server/RemoteServicesProperties.java:30]: Tbh I dont really understand why this class exist, shouldn't this code be in your new RemoteServices class instead ?

File: src/main/java/org/gridsuite/study/server/service/RemoteServices.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * @author David Braquart <david.braquart at rte-france.com>
+ */
+@Service
+public class RemoteServices {
```
Comment (flomillot) [inline src/main/java/org/gridsuite/study/server/service/RemoteServices.java:42]: I know it's ugly but maybe rename into RemoveServicesService ?

File: src/main/java/org/gridsuite/study/server/service/RemoteServices.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                .setConnectTimeout(Duration.ofMillis(REQUEST_TIMEOUT_IN_MS))
+                .setReadTimeout(Duration.ofMillis(REQUEST_TIMEOUT_IN_MS))
+                .build();
+        this.infoEndpoint = infoEndpoint;
+        this.asyncSelf = asyncRemoteServices;
+    }
+
+    public List<ServiceStatusInfos> getOptionalServices() {
+        // parallel health status check for all services marked as "optional: true" in application.yaml
+        List<CompletableFuture<ServiceStatusInfos>> results = remoteServicesProperties.getServices().stream()
```
Comment (flomillot) [inline src/main/java/org/gridsuite/study/server/service/RemoteServices.java:72]: Why don't we use `parallelStream` instead ?

File: src/test/java/org/gridsuite/study/server/service/RemoteServicesTest.java [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+    @Autowired
+    private RemoteServices remoteServices;
+
+    private static final JsonNode EMPTY_OBJ = JsonNodeFactory.instance.objectNode();
+
+    @MockBean
+    private InfoEndpoint infoEndpoint;
+
+    //@Autowired
```
Comment (flomillot) [inline src/test/java/org/gridsuite/study/server/service/RemoteServicesTest.java:74]: To remove ?

File: src/main/java/org/gridsuite/study/server/RemoteServicesProperties.java [hunk 5]
```diff
@@ -4,33 +4,50 @@
...
 import org.springframework.stereotype.Component;
+import org.springframework.validation.annotation.Validated;
 
 import java.util.List;
 import java.util.Objects;
 
 /**
  * @author David Braquart <david.braquart at rte-france.com>
  */
+@Validated
```
[the remark this answers is not in the document]
Comment (Tristan-WorkGH) [reply src/main/java/org/gridsuite/study/server/RemoteServicesProperties.java:24]: to do validation of configuration value with JSR303/349, like `@NotBlank`

File: src/main/java/org/gridsuite/study/server/RemoteServicesProperties.java [hunk 6]
```diff
@@ -4,33 +4,50 @@
...
 @Component
 @ConfigurationProperties(prefix = "gridsuite")
```
[the remark this answers is not in the document]
Comment (Tristan-WorkGH) [reply src/main/java/org/gridsuite/study/server/RemoteServicesProperties.java:26]: It's in yaml properties to say it's all in  `gridsuite` property
```yaml
#...
gridsuite:
    # anything here is mapped in RemoteServicesProperties
#...
```

File: src/main/java/org/gridsuite/study/server/RemoteServicesProperties.java (hunk 1 above)
Comment (Tristan-WorkGH) [reply src/main/java/org/gridsuite/study/server/RemoteServicesProperties.java:30]: No, in Spring we map configuration in a class to store and manage it.

File: src/main/java/org/gridsuite/study/server/service/RemoteServices.java (hunk 2 above)
Comment (Tristan-WorkGH) [reply src/main/java/org/gridsuite/study/server/service/RemoteServices.java:42]: ugly yes... dunno

File: src/main/java/org/gridsuite/study/server/service/RemoteServices.java (hunk 3 above)
Comment (Tristan-WorkGH) [reply src/main/java/org/gridsuite/study/server/service/RemoteServices.java:72]: because we can't manage the thread pool used by streams (pool size, etc...)

File: src/main/java/org/gridsuite/study/server/RemoteServicesProperties.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (flomillot) [reply src/main/java/org/gridsuite/study/server/RemoteServicesProperties.java:24]: I mean I didn't understand why it's here, then I found this article https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config.typesafe-configuration-properties.validation

File: src/main/java/org/gridsuite/study/server/RemoteServicesProperties.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (flomillot) [reply src/main/java/org/gridsuite/study/server/RemoteServicesProperties.java:26]: Yes I don't know why I didn't find the gridsuite props at first, my bad

File: src/main/java/org/gridsuite/study/server/service/RemoteServices.java (hunk 3 above)
Comment (flomillot) [reply src/main/java/org/gridsuite/study/server/service/RemoteServices.java:72]: Where is it done here ? I don't see it. We don't use StudyServerExecutionService anyway right ?
Btw I think we can, even if it's not recommend : https://www.baeldung.com/java-when-to-use-parallel-stream#3-custom-thread-pool

File: src/main/java/org/gridsuite/study/server/RemoteServicesProperties.java [hunk 7]
```diff
@@ -4,30 +4,44 @@
...
-        private String name;
-        private String baseUri;
-        private Boolean optional = false;
+        @NotBlank private String name;
+        @NotBlank private String baseUri;
+        private boolean optional = false;
+    }
+
+    public String getLocalUri() {
+        return "http://localhost:" + this.webSrvAppCtxt.getWebServer().getPort();
```
Comment (EstherDarkish) [inline src/main/java/org/gridsuite/study/server/RemoteServicesProperties.java:44]: Why is this here ?

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 8]
```diff
@@ -72,7 +70,7 @@ public class StudyController {
     private final VoltageInitService voltageInitService;
     private final LoadFlowService loadflowService;
     private final CaseService caseService;
-    private final ActuatorHealthService actuatorHealthService;
+    private final RemoteServices remoteServices;
```
Comment (EstherDarkish) [inline src/main/java/org/gridsuite/study/server/StudyController.java:73]: After discutions with team lead, let's rename this to RemoteServicesMonitor instead.

File: src/main/java/org/gridsuite/study/server/service/RemoteServices.java (hunk 2 above)
Comment (EstherDarkish) [reply src/main/java/org/gridsuite/study/server/service/RemoteServices.java:42]: After discutions with team lead, let's rename this to RemoteServicesMonitor instead.

File: src/main/java/org/gridsuite/study/server/dto/ServiceStatusInfos.java [hunk 9]
```diff
@@ -0,0 +1,15 @@
+package org.gridsuite.study.server.dto;
```
Comment (EstherDarkish) [inline src/main/java/org/gridsuite/study/server/dto/ServiceStatusInfos.java:1]:
```suggestion
/*
 * Copyright (c) 2023, RTE (http://www.rte-france.com)
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
package org.gridsuite.study.server.dto;
```

File: src/main/java/org/gridsuite/study/server/service/Inspector.java [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+import java.net.URI;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * @author David Braquart <david.braquart at rte-france.com>
```
Comment (EstherDarkish) [inline src/main/java/org/gridsuite/study/server/service/Inspector.java:39]: David or Tristan ?

Comment (Tristan-WorkGH) [reply src/main/java/org/gridsuite/study/server/service/Inspector.java:39]: David

File: src/main/java/org/gridsuite/study/server/RemoteServicesProperties.java (hunk 7 above)
Comment (Tristan-WorkGH) [reply src/main/java/org/gridsuite/study/server/RemoteServicesProperties.java:44]: forget partial clean

**Review by EstherDarkish — APPROVED** [review]
Code OK Tests OK

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 11]
```diff
@@ -1707,9 +1707,9 @@ public ResponseEntity<Void> invalidateShortCircuitStatus(@Parameter(description
     @Operation(summary = "Get the informations of all backend servers")
     @ApiResponses(value = {
         @ApiResponse(responseCode = "200", description = "The informations on all known servers"),
-        @ApiResponse(responseCode = "207", description = "Partial result because some servers haven't respond or throw an error"),
-        @ApiResponse(responseCode = "424", description = "All requests have failed, no informations retreive")})
+        @ApiResponse(responseCode = "207", description = "Partial result because some servers haven't responded or threw an error"),
+        @ApiResponse(responseCode = "424", description = "All requests have failed, no informations retrieve")})
```
Comment (flomillot) [inline src/main/java/org/gridsuite/study/server/StudyController.java:1711]:
```suggestion
        @ApiResponse(responseCode = "424", description = "All requests have failed, no informations retrieved")})
```

**Review by flomillot — APPROVED** [review]
Code OK, one trad to fix
